chore: render all examples in ws js generated readme

KunalNasa opened the pull request:
<img width="1100" alt="Screenshot 2025-05-02 at 8 05 09 PM" src="https://github.com/user-attachments/assets/7adc7873-20be-4048-a361-a228d40defc9" />

Render all examples instead of just the first one.

Fixes #1522

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Now displays multiple message examples instead of just one, allowing users to view all available examples for a given operation.

- **Bug Fixes**
	- Improved handling of cases with missing messages or examples, preventing display issues.

- **Tests**
	- Added comprehensive tests to ensure correct rendering of multiple message examples under various scenarios.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

derberg reviewed (commented):
`MessageExamples` is not tested at the moment, so we do not know if this code works or not.

I suggest you add `MessageExamples` component dedicated tests under test/components

test cases:
- operation has no messages defined = no example should be rendered
- there are no message examples = no example should be rendered
- there is one example = one example is rendered
- there are multiple examples = all are rendered
- there are multiple messages, some with example some without = all available should be rendered

KunalNasa commented:
@derberg, added tests for `MessageExamples` components.
This is the only good solution that came to my mind for writing tests for this component, and it covers all the test cases you mentioned. 
Please provide feedback for tests.

Also, added the `key` prop in  `asyncAPI React's Text` component in `MessageExamples component`. Earlier I didn't add this to keep code clean, as I thought this wouldn't lead to any optimizations, but I was getting warnings at several places, so I added it to maintain consistency.

derberg reviewed (commented):
no no please remove `key`. This is one thing rabbit tries to constantly complain on. We do not do frontend here, key's not needed

and regarding test - please adjust to follow https://github.com/asyncapi/generator/blob/master/packages/templates/clients/websocket/javascript/test/components/FileHeaderInfo.test.js so we use real fixture instead of mocked  data. Such test are more reliable - no temptation to write a mock data inside test that make test pass

KunalNasa commented:
Thank you for the suggestions. I will remove the key prop.

Regarding the tests, I understand the preference for using real fixtures instead of mocked data. However, the current real fixture doesn't seem to cover all five of the suggested test cases (cases with no messages, no examples, or mixed message states).

Would you recommend extending the existing fixture or creating a separate one specifically designed to cover these scenarios? Or should I just append a snapshot test similar to `Fileheader.test.js` using current fixture at the end of  'MessageExamples.test.js` file? This way we can have both (use a real fixture and cover test cases with mocked data).

derberg reviewed (approved):
looks great!

also this component is another example of component where we can really have it generic for any language and do something like https://github.com/asyncapi/generator/pull/1519

but yeah, separate issue and PR

derberg commented:
/rtm

KunalNasa commented:
> looks great!
> 
> also this component is another example of component where we can really have it generic for any language and do something like #1519
> 
> but yeah, separate issue and PR

Hi @derberg, we already have a separate issue opened for this.  
Please have a look at #1524 issue's discussion. I have dropped some suggestions regarding the refactoring of components. Would love to hear your thoughts as well. 😊
